ir/src/symbol_table/symbol_type.rs

```diff
@@ -4,7 +4,7 @@ use std::fmt::Display;
 #[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
 pub(crate) enum SymbolType {
     /// an identifier for a constant, containing its type and value
-    Constant(ConstantValueExpr),
+    ConstantBinding(ConstantValueExpr),
     /// an identifier for a binding to one or more trace columns, containing the trace binding
     /// information with its identifier, trace segment, size, and offset.
     TraceBinding(TraceBinding),
```

Comment:
I'm wondering if a better name for `SymbolType` is actually `SymbolBinding` (since it actually defines what the symbol is bound to). I would potentially redefine `Symbol` struct as:
```Rust
pub(crate) struct Symbol {
    name: String,
    binding: SymbolBinding,
}
```
And then rename `SymbolType` into `SymbolBinding` and keep the enum variant names simple. Maybe something like:
```Rust
pub(crate) enum SymbolType {
    Constant(ConstantValueExpr),
    Trace(TraceBinding),
    PublicInput(usize),
    PeriodicColumn(usize, usize),
    Variable(VariableValueExpr),
    RandomValues(usize, usize),
}
```